ReactiveCocoaTests/Swift/SignalSpec.swift

```diff
@@ -453,6 +453,12 @@ class SignalSpec: QuickSpec {
 				observer.sendNext(1)
 				expect(lastValue).to(equal("2"))
 			}
+
+			it("should not keep resulting signal alive indefinitely") {
+				weak var signal: Signal<AnyObject, NoError>? = Signal.never.map { $0 }
```

Comment:
@NachoSoto But what you describe isn't exactly the case of #2585, which is about a running signals that neither deallocates nor emit terminating events.

The issue can also happen on normal signals, if `takeUntil` happens later than a `map`, so that the derived signal of the mapped signals ends while the mapped signal is still active.

Nonetheless, it seems another boundary case that may need a fix.